robocop/checkers/misc.py

```diff
@@ -270,34 +269,49 @@ class ReturnChecker(VisitorChecker):
 
     reports = (
         "keyword-after-return",
-        "keyword-after-return-from",
         "empty-return",
     )
 
     def visit_Keyword(self, node):  # noqa
         return_setting_node = None
         keyword_after_return = False
         return_from = False
+        error = ""
         for child in node.body:
             if isinstance(child, Return):
                 return_setting_node = child
+                error = (
+                    "[Return] is not defined at the end of keyword. "
+                    "Note that [Return] does not return from keyword but only set returned variables"
+                )
                 if not child.values:
                     self.report("empty-return", node=child, col=child.end_col_offset)
-            elif isinstance(child, KeywordCall):
+            elif ReturnStatement and isinstance(child, ReturnStatement):
+                return_setting_node = child
+                error = "RETURN is not defined at the end of keyword"
+            elif not isinstance(child, (EmptyLine, Comment)):
+                if return_setting_node is not None:
+                    keyword_after_return = True
+
+            if isinstance(child, KeywordCall):
                 if return_setting_node is not None:
                     keyword_after_return = True
                 if return_from:
                     token = child.data_tokens[0]
                     self.report(
-                        "keyword-after-return-from",
+                        "keyword-after-return",
+                        error_msg="Keyword call after 'Return From Keyword' keyword",
```

Comment:
I think we can safely remove this redundant word here:
```suggestion
                        error_msg="Keyword call after 'Return From Keyword'",
```